### PR TITLE
EUI-3348 - Default organisation selection

### PR DESCRIPTION
**Pull request by leepavlouhmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3348

### Change description ###
preselect organisation address on case create

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/shared/components/palette/organisation/write-organisation-field.component.html [hunk 1]
```diff
@@ -43,7 +43,7 @@ <h2 class="heading-h2">Organisation name and address</h2>
               <ccd-markdown [content]="organisation.address"></ccd-markdown>
             </td>
             <td class="td-select">
-              <a *ngIf="(selectedOrg$ | async).organisationIdentifier === ''" href="javascript:void(0);" (click)="selectOrg(organisation)" title="Select the organisation {{organisation.name}}">Select</a>
+              <a *ngIf="!(selectedOrg$ | async)?.organisationIdentifier" href="javascript:void(0);" (click)="selectOrg(organisation)" title="Select the organisation {{organisation.name}}">Select</a>
```
Comment (paul-graham) [inline src/shared/components/palette/organisation/write-organisation-field.component.html:46]: This `*ngIf` isn't needed at all as it's nested inside of an identical one above.

File: src/shared/components/palette/organisation/write-organisation-field.component.html [hunk 2]
```diff
@@ -54,7 +54,7 @@ <h2 class="heading-h2">Organisation name and address</h2>
           </div>
         </div>
       </div>
-      <div *ngIf="(selectedOrg$ | async).organisationIdentifier">
+      <div *ngIf="(selectedOrg$ | async)?.organisationIdentifier">
         <table id="organisation-selected-table" *ngIf="(selectedOrg$ | async) as selectedOrg">
```
Comment (paul-graham) [inline src/shared/components/palette/organisation/write-organisation-field.component.html:58]: Again, this is covered by the `*ngIf` in the line directly above.

File: src/shared/components/palette/organisation/write-organisation-field.component.html [hunk 3]
```diff
@@ -54,7 +54,7 @@ <h2 class="heading-h2">Organisation name and address</h2>
           </div>
         </div>
       </div>
-      <div *ngIf="(selectedOrg$ | async).organisationIdentifier">
+      <div *ngIf="(selectedOrg$ | async)?.organisationIdentifier">
```
Comment (paul-graham) [inline src/shared/components/palette/organisation/write-organisation-field.component.html:57]: _Suggestion:_
You could wrap this in a template and set line 37 to have an else:
`<div *ngIf="!(selectedOrg$ | async)?.organisationIdentifier; else selectedOrganisation" ...>`

And then:
```
<ng-template #selectedOrganisation>
  <table id="organisation-selected-table" ...>
    ...
  </table>
</ng-template>
```

File: src/shared/components/palette/organisation/write-organisation-field.component.ts [hunk 4]
```diff
@@ -29,8 +35,11 @@ export class WriteOrganisationFieldComponent extends AbstractFieldWriteComponent
   public simpleOrganisations$: Observable<SimpleOrganisationModel[]>;
   public selectedOrg$: Observable<SimpleOrganisationModel>;
 
-  constructor(private organisationService: OrganisationService, private organisationConverter: OrganisationConverter) {
+  constructor(private organisationService: OrganisationService,
```
Comment (paul-graham) [inline src/shared/components/palette/organisation/write-organisation-field.component.ts:38]: All of these parameters should be `readonly`.

File: src/shared/components/palette/organisation/write-organisation-field.component.ts [hunk 5]
```diff
@@ -41,20 +50,43 @@ export class WriteOrganisationFieldComponent extends AbstractFieldWriteComponent
...
     this.organisationFormGroup = this.registerControl(new FormGroup({}), true) as FormGroup;
-    if (this.caseField && this.caseField.value && this.caseField.value.OrganisationID) {
-      this.instantiateOrganisationFormGroup(this.caseField.value.OrganisationID, this.caseField.value.OrganisationName);
-      this.selectedOrg$ = this.organisations$.pipe(
-        map(organisations =>
-          organisations.filter(findOrg => findOrg.organisationIdentifier === this.caseField.value.OrganisationID)
-                       .map(organisation => this.organisationConverter.toSimpleOrganisationModel(organisation))[0]),
-      );
-      this.searchOrgTextFormControl.disable();
+    if (this.parent.controls && this.parent.controls.hasOwnProperty(WriteOrganisationFieldComponent.PRE_POPULATE_TO_USERS_ORGANISATION)
```
Comment (paul-graham) [inline src/shared/components/palette/organisation/write-organisation-field.component.ts:53]: This is a _big_ `if` statement!

File: src/shared/components/palette/organisation/write-organisation-field.component.ts [hunk 6]
```diff
@@ -29,8 +35,11 @@ export class WriteOrganisationFieldComponent extends AbstractFieldWriteComponent
...
 
-  constructor(private organisationService: OrganisationService, private organisationConverter: OrganisationConverter) {
+  constructor(private organisationService: OrganisationService,
+              private organisationConverter: OrganisationConverter,
+              private windowService: WindowService) {
     super();
+    this.defaultOrg = JSON.parse(this.windowService.getSessionStorage(WriteOrganisationFieldComponent.ORGANISATION_DETAILS));
   }
 
   public ngOnInit() {
```
Comment (paul-graham) [inline src/shared/components/palette/organisation/write-organisation-field.component.ts:45]: Needs a `: void` return type.

**Review by paul-graham — CHANGES_REQUESTED** [review]
Just a couple of things that could with a little bit of tidying up, I think. And it should be pointing at `Release`, not `master`.

File: src/shared/components/palette/organisation/write-organisation-field.component.html (hunk 1 above)
Comment (rayliangatos) [reply src/shared/components/palette/organisation/write-organisation-field.component.html]: good spot, removed

File: src/shared/components/palette/organisation/write-organisation-field.component.html (hunk 2 above)
Comment (rayliangatos) [reply src/shared/components/palette/organisation/write-organisation-field.component.html:58]: this is diferent level

File: src/shared/components/palette/organisation/write-organisation-field.component.html (hunk 3 above)
Comment (rayliangatos) [reply src/shared/components/palette/organisation/write-organisation-field.component.html]: took your advice, thanks.

File: src/shared/components/palette/organisation/write-organisation-field.component.ts (hunk 4 above)
Comment (rayliangatos) [reply src/shared/components/palette/organisation/write-organisation-field.component.ts]: done

File: src/shared/components/palette/organisation/write-organisation-field.component.ts (hunk 5 above)
Comment (rayliangatos) [reply src/shared/components/palette/organisation/write-organisation-field.component.ts:53]: yes indeed

File: src/shared/components/palette/organisation/write-organisation-field.component.ts (hunk 6 above)
Comment (rayliangatos) [reply src/shared/components/palette/organisation/write-organisation-field.component.ts]: changed

**Review by paul-graham — DISMISSED** [review]
Thanks for making the request changes, @rayliangatos. 👍

**Review by Mo-Lala84 — DISMISSED** [review]
approved

**Review by paul-graham — APPROVED** [review]

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Pull request merged**